### DISCUSSION TITLE
Nerf or remove Air Lock/Cloud Nine

### DISCUSSION THREAD

**Discussion by rglezg** [Renegade-Engine/Renegade-Emerald#17] [Ideas]
Continuation of #11.
Now that all battles will be doubles, Rayquaza's signature ability Air Lock could be a problem. I think that, as the version's legendary, it deserves such ability but maybe is too overpowered for the new battle system.

**Comment by DP-404** [comment]
> 
> 
> Continuation of #11.
> Now that all battles will be doubles, Rayquaza's signature ability Air Lock could be a problem. I think that, as the version's legendary, it deserves such ability but maybe is too overpowered for the new battle system.

You may nerf Air Lock: every turn Rayquaza is in the battlefield, there is a chance of negating the effects of weather.

**Reply by rglezg** [reply]
That could be a good solution, but may be tricky to implement that for a whole turn. The macro for checking if the weather has effect is defined in battle_util.h:34
`#define WEATHER_HAS_EFFECT ((!IsAbilityOnField(ABILITY_CLOUD_NINE) && !IsAbilityOnField(ABILITY_AIR_LOCK)))`

and is called every time the game calculates something related to weather. So in case we change it to depend on a random value, we have 2 options:

1. Call the macro once per turn and store its result in a new field on gBattleStruct.
2. Only change the macro so the chance of negating the effect will be calculated independently every time.

**Note that the macro also checks for the ability Cloud Nine, that haves the same effect as Air Lock and should get the same treatment.**

**Reply by SoyNegro** [reply]
Need to test it up...(after downloading compiler again) but it looks like this 

  ```
#define CLOUD_NINE_PROC ((Random()%5)<3)
  #define AIR_LOCK_PROC  ((Random()%10)<9)
  #define WEATHER_HAS_EFFECT ((!ABILITY_ON_FIELD(ABILITY_CLOUD_NINE) && CLOUD_NINE_PROC ) &&   (!ABILITY_ON_FIELD(ABILITY_AIR_LOCK)&& AIR_LOCK_PROC ))
  #define WEATHER_HAS_EFFECT2 ((!ABILITY_ON_FIELD2(ABILITY_CLOUD_NINE) && CLOUD_NINE_PROC ) && (!ABILITY_ON_FIELD2(ABILITY_AIR_LOCK)&& AIR_LOCK_PROC ))
```

Like that Cloud Nine will have a 60% proc of cancelling weather effects while  Air Lock 90%.
Moves that depend of weather like Sun Beam will anyways lose effects. But with this arrangement Castform may transform sometimes if battling such pokemons, besides weathers conditions affecting moves may also apply.

**Reply by rglezg** [reply]
That's what I was thinking. The probabilities can be lowered later if needed. But why you proposed two macros that are exactly the same?

> Moves that depend of weather like Sun Beam will anyways lose effects.

Why? I thought the goal here was to make moves like Solar Beam, Synthesis, Thunder, Blizzard, Weather Ball, etc depend on that probability.

**Comment by SoyNegro** [comment]
Even though we may be soon looking in-depth to the Ability List, this two abilities are without doubt to OP for doubles battles. But since the list of pokemon with these abilities is short we may in turn put it as a random hidden ability of them. We can make that getting Cloud Nine as ability depend on chance. A very small one.

**Reply by rglezg** [reply]
So I suggest making both abilities work with a random chance, giving Cloud Nine a lower probability than Air Lock for keeping Rayquaza special :)

**Reply by rglezg** [reply]
Would you like to implement this, @SoyNegro?

**Comment by SoyNegro** [comment]
It had to be tested. Moves depending on weather should work the turn in
which Cloud Nine or Air Lock arent proc but Solar Beam itself as a charging
move may not work. Further testing is needed.

**Reply by rglezg** [reply]
The check for Solar Beam is made only once: in the turn you tell your Pokémon to use it. If it bypass Cloud Nine/Air Lock, will be fired instantly. On the other hand, if the weather effects are blocked, it will charge one turn, and fired the next as normal. If is fired as a 2-turn move, the game doesn't check if weather has effect the second turn, it makes no sense.
